src/components/TodoList.jsx

```diff
@@ -13,51 +13,121 @@ class TodoListXP extends React.Component {
+        this.removeAllTasks = this.removeAllTasks.bind(this)
+        this.sortTasks = this.sortTasks.bind(this)
     }
 
-    removeTask(text) {
-        const { tasks } = this.state;
 
+    sortTasks(event) {
+        const { tasks } = this.state
+        const option = event.target.value
+        let newState
+        if (option === "Name asc") {
+            newState = {
+                ...this.state,
+                tasks: tasks.sort((task1, task2) => (task1.taskName.toLowerCase() > task2.taskName.toLowerCase()) ? 1 : -1)
+            }
+        }
+        if (option === "Name desc") {
+            newState = {
+                ...this.state,
+                tasks: (tasks.sort((task1, task2) => (task1.taskName.toLowerCase() > task2.taskName.toLowerCase()) ? 1 : -1)).reverse()
+            }
+        }
+        if (option === "Newer") {
+            newState = {
+                ...this.state,
+                tasks: (tasks.sort((task1, task2) => (task1.date - task2.date))).reverse()
+            }
+        }
+        if (option === "Older") {
+            newState = {
+                ...this.state,
+                tasks: tasks.sort((task1, task2) => (task1.date - task2.date))
+            }
+        }
 
-        const newState = {
+        this.setState(newState)
+    }
+
+
+    taskTexts = () => {
+        const taskNames = []
+        for (const task of this.state.tasks) {
+            taskNames.push(task.taskName)
+        }
+        return taskNames
+    }
+
+
+    changeText = (text, newText) => {
+        const { tasks } = this.state
+        const id = this.taskTexts().indexOf(text)
+        tasks[id].taskName = newText
+        this.setState({
             ...this.state,
-            tasks: tasks.filter(elem => elem != text)
+            tasks: tasks
+        })
+    }
+
+
+    removeTask(text) {
+        const { tasks } = this.state
+        tasks.splice(this.taskTexts().indexOf(text), 1)
+        const newState = {
+            ...this.state
         }
         this.setState(newState);
     }
 
     addTask() {
         const { tasks } = this.state;
-
         this.setState({
             input: "",
-            tasks: tasks.concat(this.state.input)
+            tasks: tasks.concat({ taskName: this.state.input, date: new Date() })
         });
     }
 
+
+
     onInputChange(event) {
         const text = event.target.value;
-        this.setState({ ...this.state, input: text });
+        this.setState({ ...this.state, input: text })
+    }
+
+    removeAllTasks() {
+        this.setState(
+            {
+                tasks: []
+            }
+        )
     }
 
     render() {
         return <div>
             <div>
+                <select onChange={this.sortTasks}>
+                    <option>Sort by</option>
+                    <option>Newer</option>
+                    <option>Older</option>
+                    <option>Name asc</option>
+                    <option>Name desc</option>
```

Comment:
Many elements in a list written by hand. Do you see how this could (and should) be automated using an array of strings and `map()` inside JSX? :-) Let's do it.